nameserver/zone.go

```diff
@@ -3,151 +3,761 @@ package nameserver
 import (
 	"bytes"
 	"fmt"
+	"github.com/benbjohnson/clock"
 	"github.com/miekg/dns"
 	. "github.com/weaveworks/weave/common"
 	"net"
 	"sync"
+	"time"
 )
 
 const (
 	RDNSDomain = "in-addr.arpa."
 )
 
+const (
+	DefaultLocalDomain     = "weave.local." // The default name used for the local domain
+	DefaultRefreshInterval = int(localTTL)  // Period for background updates with mDNS
+	DefaultRelevantTime    = 60             // When to forget info about remote info if nobody asks...
+
+	defaultNumUpdaters    = 1              // Number of background updaters
```

Comment:
Yes, `InsistentLookupName` is a blocking operation, but I'm not sure what the right number would be... `InsistentLookupName` takes 500ms, so we can refresh 2 names per second... maybe I should increase this number, set some low, safe value (4 workers?) and add a command line argument...